app/services/forest_liana/permissions_getter.rb

```diff
@@ -1,29 +1,22 @@
 module ForestLiana
-  class PermissionsGetter < ForestApiRequester
+  class PermissionsGetter
     def initialize(rendering_id)
-      @uri = URI.parse("#{forest_api_url}/liana/v2/permissions")
+      @route = "/liana/v2/permissions"
       @rendering_id = rendering_id
     end
 
     def perform
-      perform_request({ 'renderingId' => @rendering_id })
-    rescue => exception
-      FOREST_LOGGER.error "Cannot retrieve the permissions for the project you\'re trying to unlock. Forest API seems to be down right now."
-      FOREST_LOGGER.error exception
-      nil
-    end
-
-    private
+      begin
+        query_parameters = { 'renderingId' => @rendering_id }
+        response = ForestApiRequester.get(@route, query: query_parameters)
 
-    def handle_service_response(response)
-      if response.is_a?(Net::HTTPOK)
-        JSON.parse(response.body)
-      elsif response.is_a?(Net::HTTPNotFound) || response.is_a?(Net::HTTPUnprocessableEntity)
-        FOREST_LOGGER.error "Cannot retrieve the permissions from the Forest server. Can you check that you properly set up the forest_env_secret?"
-        nil
-      else
-        FOREST_LOGGER.error "Cannot retrieve the data from the Forest server. An error occured in Forest API."
-        nil
+        if response.is_a?(Net::HTTPOK)
+          JSON.parse(response.body)
+        else
+          raise "Cannot retrieve the data from the Forest server. An error occured in Forest API."
+        end
+      rescue
```

Comment:
The original exception (the one which have the message) will be encapsulated to the `HTTP401Error` exception in a `self.cause` attribute. This is a built in feature in ruby since version 2.1.